retrievalmarket/impl/providerstates/provider_states.go

```diff
@@ -30,7 +30,23 @@ type ProviderDealEnvironment interface {
 }
 
 func firstSuccessfulUnseal(ctx context.Context, node rm.RetrievalProviderNode, pieceInfo piecestore.PieceInfo) (io.ReadCloser, error) {
+
+	// prefer an unsealed sector containing the piece if one exists
+	for _, deal := range pieceInfo.Deals {
+		isUnsealed, err := node.IsUnsealed(ctx, deal.SectorID, deal.Offset.Unpadded(), deal.Length.Unpadded())
+		if err != nil {
+			continue
```

Comment:
Let's return the last error if it's not possible to find any sectors with the piece